src/python/py/models/builder.py

```diff
@@ -413,16 +415,74 @@ def save_model(self, out_dir):
         )
 
     def to_int4(self, model):
-        quant = MatMul4BitsQuantizer(
+        if (self.quant_provider == "nvidia_awq"):
+            # Nvidia ModelOpt only accepts an ONNX model path as input so process is
+            # hf --> fp16 model.onnx file (store at temp) --> pass fp16 model.onnx to modelopt
+            import tempfile
+            from onnxruntime.quantization.matmul_4bits_quantizer import NVAWQWeightOnlyQuantConfig
+            out_dir = os.path.join(tempfile.gettempdir(), "temp")
+            os.makedirs(out_dir, exist_ok=True)
+
+            out_path = os.path.join(out_dir, self.filename)
+            data_path = os.path.join(out_dir, os.path.basename(out_path) + ".data")
+
+            if os.path.exists(out_path):
+                print(f"Overwriting {out_path}")
+                os.remove(out_path)
+            if os.path.exists(data_path):
+                print(f"Overwriting {data_path}")
+                os.remove(data_path)
+
+            try:
+                save_model(
+                    model,
+                    out_path,
+                    save_as_external_data=True,
+                    all_tensors_to_one_file=True,
+                    location=os.path.basename(data_path),
+                    size_threshold=0,
+                    convert_attribute=False,
+                )
+
+                if self.nvidia_awq_calibration == "awq":
+                    calibration_method = "awq_lite"
+                elif self.nvidia_awq_calibration == "awq_clip":
+                    calibration_method = "awq_clip"
+                else:
+                    assert False, f"Unexpected calibration method: {self.nvidia_awq_calibration}"
+                print("nvidia_awq quantization works with block_size 128")
+                quant_config = NVAWQWeightOnlyQuantConfig(
+                    tokenizer_dir=self.model_name_or_path,
+                    calibration_method=calibration_method,
+                )
+
+                quant = MatMul4BitsQuantizer(
+                    model=out_path,
```

Comment:
We should simplify this method's logic since both paths will create a `MatMul4BitsQuantizer`, run `quant.process()`, and return `quant.model.model`.